sfdoc/publish/tasks.py

```diff
@@ -10,294 +10,368 @@
 
 from .amazon import S3
 from .exceptions import SfdocError
-from .html import HTML
+from .html import HTML, collect_html_paths
 from .logger import get_logger
 from .models import Article
 from .models import Bundle
 from .models import Image
 from .models import Webhook
 from .salesforce import Salesforce
-from .logger import get_logger
-from .utils import is_html
-from .utils import skip_html_file
-from .utils import unzip
+from . import utils
+
+logger = None
+
+AWS_S3_PUBLISHED_HTML_REPOSITORY_URL = (
+    f"s3://{settings.AWS_S3_BUCKET}/{settings.AWS_S3_PUBLISHED_HTML_REPOSITORY_DIR}"
+)
+
+AWS_S3_DRAFT_HTML_REPOSITORY_URL = (
+    f"s3://{settings.AWS_S3_BUCKET}/{settings.AWS_S3_DRAFT_HTML_REPOSITORY_DIR}"
+)
+
+
+def _download_and_unpack_easydita_bundle(bundle, path):
+    logger.info("Downloading easyDITA bundle from %s", bundle.url)
+    auth = (settings.EASYDITA_USERNAME, settings.EASYDITA_PASSWORD)
+    response = requests.get(bundle.url, auth=auth)
```

Comment:
Should we assert that it's an https URL so we aren't sending credentials in plaintext?

Reply:
b11bf8bd6c7e097f736262dcb53916e8d53b1ed2